core/src/main/java/macrobase/analysis/summary/itemset/RiskRatio.java

```diff
@@ -0,0 +1,62 @@
+package macrobase.analysis.summary.itemset;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RiskRatio {
+    private static final Logger log = LoggerFactory.getLogger(RiskRatio.class);
+
+    private static double computeDouble(double exposedInlierCount,
+                                        double exposedOutlierCount,
+                                        double totalInliers,
+                                        double totalOutliers) {
+        double totalExposedCount = exposedInlierCount + exposedOutlierCount;
+        double unexposedOutlierCount = (totalOutliers - exposedOutlierCount);
+        double totalMinusExposedCount = totalInliers + totalOutliers - totalExposedCount;
+
+        // no exposure occurred
+        if (totalExposedCount == 0) {
+            log.error("Computing risk ratio with no exposure.");
+            return 0;
+        }
+
+        // we only exposed this ratio, everything matched!
+        if (totalMinusExposedCount == 0) {
+            return 0;
```

Comment:
One common solution to zero events seems to be adding 0.5 to all cell values (e.g. http://www.ncbi.nlm.nih.gov/pubmed/8085071). I think it might be worth spending a little time thinking about the edge cases.